### PR TITLE
Extract build-info from traces

### PR DESCRIPTION
**Pull request by lasote**
#942

### CODE REVIEW

File: conans/build_info/conan_build_info.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+def _extract_uploads_from_conan_trace(path):
+    modules = defaultdict(dict)  # dict of {conan_ref: [abs_path1, abs_path2]}
+    try:
+        with open(path, "r") as traces:
+            for line in traces.readlines():
+                doc = json.loads(line)
+                if doc["_action"] in ("UPLOADED_RECIPE", "UPLOADED_PACKAGE"):
+                    module_type = "recipe" if doc["_action"] == "UPLOADED_RECIPE" else "package"
+                    modules[doc["_id"]] = {"remote": doc["remote"], "files": [], "type": module_type}
+                    for file_doc in doc["files"]:
```
Comment (memsharded) [inline conans/build_info/conan_build_info.py]: oneliner ``modules[doc["_id"]]["files"].extend(doc["files"])``

File: conans/build_info/model.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import conans
+
+
+class BuildInfo(object):
+
+    def __init__(self, name=None, number=None, started=None):
```
Comment (memsharded) [inline conans/build_info/model.py]: unused args in constructor

File: conans/build_info/model.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        self.artifacts = []
+        self.dependencies = []
+
+    def serialize(self):
+        return {"id": self.id,
+                "artifacts": [ar.serialize() for ar in self.artifacts],
+                "dependencies": [dep.serialize() for dep in self.dependencies]}
+
+
+class BuildInfoModuleArtifact(object):
```
Comment (memsharded) [inline conans/build_info/model.py]: why not namedtuples?

Comment (lasote) [inline conans/build_info/model.py]: When I had these code in package tools I did more things like send to artifactory and some other processing, but now, sure, only named tuples.

**Pull request merged**